src/file.h

```diff
@@ -0,0 +1,48 @@
+#ifndef REDISRAFT_FILE_H
+#define REDISRAFT_FILE_H
+
+#include <stdlib.h>
+#include <sys/types.h>
+
+/* File implementation with a userspace buffer.
+ *
+ * Main differences over the standard library FILE implementation:
+ *
+ * * Works with O_APPEND mode only. It is a deliberate decision to keep the
+ *   implementation simple.
+ * * The standard library FILE API requires extra care when you want to use it
+ *   together with POSIX IO functions, e.g., need fflush() before ftruncate().
+ *   This implementation tries to do these things automatically.
+ * * Tracks read/write offset internally. Getting read/write offset is fast
+ *   as it doesn't make a system call for them.
+ * * This implementation is not thread-safe while the standard library
+ *   implementation is.
+ */
+
+typedef struct File {
+    int fd;         /* File descriptor */
+    char *rpos;     /* In read mode, current read position of the buffer. */
+    char *rend;     /* In read mode, end position of the buffer. */
+    size_t roffset; /* Read offset relative to the head of the file. */
+    char *wpos;     /* In write mode, current write position of the buffer. */
+    size_t woffset; /* Write offset relative to the head of the file. */
+    char buf[4096]; /* Userspace buffer. */
+} File;
+
+void FileInit(File *file);
+int FileTerm(File *file);
```

Comment:
This is another topic we need discuss, hopefully over the code convention doc. 

For object initializer/destructor patterns, I was thinking something like:

```c
void FileInit(File *file);    // this will not allocate memory for `file`
void FileTerm(File *file);    // this will terminate/deinit but not call `free()` for `file`
File *FileCreate(void);       // this will allocate memory for `file` and initialize it
void FileDestroy(File* file); // this will terminate/deinit `file` object and also call `free()` for `file`
```

This is the naming pattern I used before. I'm open to suggestions but I'll be happy if we have a codebase that follows a standard way. So, all objects will follow the same pattern: `ConnCreate(), ConnDestroy()`, `ThreadPoolInit(), ThreadPoolTerm()` or both if required: `ShardGroupInit(), ShardGroupTerm(), ShardGroupCreate(), ShardGroupDestroy()`. 

The goal is to increase readability. We'll able to guess initializer function name, destructor function name, whether these functions will deallocate the memory etc. 

Maybe we merge as it is right now and go back to this topic later when we come to a conclusion over the naming?